graphql/resolvers/addComment.ts

```diff
@@ -57,7 +57,7 @@ export const addComment = async (
   const { author, submission } = comment
 
   if (author.id !== submission.user.id && submission.user.discordId) {
-    const notificationEmbed: MessageEmbedOptions = {
+    const notificationEmbed: EmbedBuilder['data'] = {
```

Comment:
Question my own understanding: is `EmbedBuilder['data']` is a type annotation, right?

Is there a reason for using bracket notation instead of dot notation (`EmbedBuilder.data`), or will it work either way?

Reply:
Indexed access is how we can look up a specific property on another type, this is a typescript feature, `EmbedBuilder.data` will not work here. https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html

`EmbedBuilder` class has a property `data`, when using `EmbedBuilder` as a type (like here) if you want to access the type of `data` property, you will have to use the indexed access, dot access will not work.

I have created this [example](https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBYKodp0b6CRqgAC8oAA3mJmJB8v12IhpN8ANz9BjIDmCVCUTjMfkMaC6dnjeDwFpeAC+vMUlKIlTZDOpP0aAG0AEQMNkGgC6qvVKAAchUtYzaDTGgFjZBVcUcM0PGQvIh4H0GMD2DQvm4zMx2C7+LBpbBEgH4HoRF4DQBRVaChb6A1JbEG8AOVAAcmEVsgEgNQRUAHkvmR3oJIPw2UdUI4FK5pHhQE5fTphlh9CgEIhocJLiGJH7oiIDBxsQsmKwODwgiUgA) as you can see that `type Age` got correctly defined as `number` but `NAge` did not.